### ISSUE TITLE
Stopped reporting unlocks by user

### ISSUE THREAD

**Issue by proisland** [TedTolboom/no.IDLock#14]
I installed version v1.2.0 of your app on Homey version 1.15.13, and setup and received notifications for users unlocking the door, but it suddenly stopped working. It also didnt start working after updating to Homey version 2.0.0. and therefor also restarting Homey and its apps. I do still receive notifications about "Button unlocked the door", so notifications do work for unlockings done from the inside, which is of course less interesting.

**Comment by mansica** [comment]
I hade the same problem. I got it working by unpairing/pairing the device. Not so fun but for some reason it didn't report all notifications. i'm not completely sure if it was due to going to version 2 of Homey or upgrading the firmware of idlock.

**Comment by proisland** [comment]
I have tried unpairing/pairing it, but in my case it didnt fix the issue. I’m running the version updated using the 1.1(8) version of the ID Lock app from TestFlight.

**Comment by mansica** [comment]
Ok, not sure what version you are referring to. I'm running the latest firmware from IDLock, my fork of IDLock(version 1.2.0 with some setting add ons) and version 2.1.1 of Homey and it is working as it should after I reinstalled the device.

**Comment by mickem** [comment]
I had the same issue, and first re-installed the app, then re-added the lock to Homey and still no notification from users.
Then a days or so later I installed a development version of the app to see if I could see whats amiss, and it all started to work.

So not sure whats causing the issue, but re-adding the lock did not fix the issue...

**Comment by proisland** [comment]
@TedTolboom - do you have any idea why it started working for @mickem after installing the dev version? I havnt tried this myself.

**Comment by mickem** [comment]
I actually have an idea, and it is a bug in the firmware of the lock, and thus it sometimes works... 
So for me it worked a few times after the software change but then stopped again...
I have a support ticket with id lock, and they said:

> We have heard this from other people too, but not anyone who are using firmware version 1.1.7.
> 
> Are you sure you have the latest firmware on the doorlock?
> If yes, please try to exclude the node from Homey, perform a Z-wave local reset and remove the Z-wave module. After removing the module factory reset the lock and then include the lock again in your system.

As I have the latest firmware and have included/excluded I will try to reset the lock and see if that resolves the issue.
If anyone else has this issue on 1.1.7 firmware please raise an issue with idlock so they are aware that I am not the only one.

So in essence, this is not a bug in the app but in the lock.
The bug is related to the notification message which is never sent when unlocking from the outside.
If you check the z-wave log you can see that the lock only reports:
```
2: unlocked from inside
1: self locked
<missing 4/6>
1: self locked

```
// Michael Medin

**Comment by proisland** [comment]
Thank you for your explenation @mickem. I have exactly the same issue as you, so I reported a ticket to ID Lock myself now hoping they will fix the issue.

**Comment by mansica** [comment]
Hi,

I have the same experience. You will however get the notification if you wait to open the door a couple of seconds after unlocking it. Otherwise it will only send the open door notification.

Hopefully idlock will fix this issue in next firmware upgrade.

/Mats

**Comment by mickem** [comment]
Ohh, that might be why I get it "sporadically" I noticed the same on the "open by button" notification. If I am to quick I did not get the notification.

**Comment by proisland** [comment]
I tested waiting for two seconds to open the door after unlocking the door, and I then received the notification for the first time in a very lon time.

**Comment by proisland** [comment]
This issue seems to have been fixed (at least for me) after updating to ID Lock Firmware version 1.4.9 released 23.08.2019. Anyone else tested using the new version?

**Comment by mickem** [comment]
I can also confirm the update seems to resolve the issue.

A word of warning: My rfid tags stopped working after this update.
